pdns/recursordist/rec-main.cc

```diff
@@ -626,31 +625,41 @@ static std::shared_ptr<std::vector<std::unique_ptr<FrameStreamLogger>>> startFra
   return result;
 }
 
+static void asyncFrameStreamLoggersCleanup(std::shared_ptr<std::vector<std::unique_ptr<FrameStreamLogger>>>&& servers)
+{
+  auto thread = std::thread([&] {
+    servers.reset();
+  });
+  thread.detach();
+}
+
 bool checkFrameStreamExport(LocalStateHolder<LuaConfigItems>& luaconfsLocal)
 {
   if (!luaconfsLocal->frameStreamExportConfig.enabled) {
-    if (t_frameStreamServers) {
+    if (t_frameStreamServersInfo.servers) {
       // dt's take care of cleanup
-      t_frameStreamServers.reset();
+      asyncFrameStreamLoggersCleanup(std::move(t_frameStreamServersInfo.servers));
```

Comment:
```suggestion
      asyncFrameStreamLoggersCleanup(std::move(t_frameStreamServersInfo.servers));
      t_frameStreamServersInfo.config = luaconfsLocal->frameStreamExportConfig;
```

Reply:
Is that change because the config was previously enabled and now it isn't and so we need to update it?

Reply:
Right. Thanks!